### PR TITLE
Add extra options

### PR DESCRIPTION
**Pull request by sergio-costas**
Allows to specify several extra options in the "extra_data" field in each DBus call. These options allow to set an icon, change the window's title, or the message.

image

In order to get access to any SNAP icon, it requires a "system-files" interface and permissions, which, for the snap store, requires specific permissions https://snapcraft.io/docs/system-files-interface

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -14,6 +14,12 @@ layout:
   /usr/share/locale:
     bind: $SNAP/usr/share/locale
 
+plugs:
+  read-snap-data:
+    interface: system-files
```
Comment (kenvandine) [inline snap/snapcraft.yaml:19]: We could use the desktop-launch interface to get access to the icon.  Even though we aren't currently using that interface to launch snaps, it does provides access to the desktop files and icons.

Comment (sergio-costas) [reply snap/snapcraft.yaml:19]: I'll test that.

Comment (sergio-costas) [reply snap/snapcraft.yaml:19]: Mmm... it's not enough... the icon paths that I receive are inside the snaps, while "desktop-launch" only gives access to /var/lib/snapd/desktop/icons/...

Comment (kenvandine) [reply snap/snapcraft.yaml:19]: The desktop-launch interface does include "/snap/*/*/** r," and I think it's more compelling to get permission to plug desktop-launch than system-files.

https://github.com/snapcore/snapd/blob/master/interfaces/builtin/desktop_launch.go#L46

Comment (sergio-costas) [reply snap/snapcraft.yaml]: Odd... it works, but I think that the last time that I tested it, it didn't... Maybe I forgot to connect the plug... Anyway, change done.

**Review by kenvandine — APPROVED** [review]
Thanks for the change to use desktop-launch, I think this is good now.

**Pull request merged**